core/src/main/java/org/apache/struts2/util/StrutsUtil.java

```diff
@@ -125,12 +120,10 @@ public Object findValue(String expression, String className) throws ClassNotFoun
 
     public Object findValue(String expr, Object context) {
         try {
-            return ognl.getValue(expr, ActionContext.getContext().getContextMap(), context);
-        } catch (OgnlException e) {
-            if (e.getReason() instanceof SecurityException) {
-                LOG.error(format("Could not evaluate this expression due to security constraints: [{0}]", expr), e);
-            }
-            return null;
+            stack.push(context);
```

Comment:
Should be outside the try block so that we don't pop unless successfully pushed

Reply:
fixed